server/services/scraperCraigslist.js

```diff
@@ -0,0 +1,39 @@
+const puppeteer = require("puppeteer");
+
+//Returns title, price, and image URL if item is still up. If the item isn't still up, returns isStillAvailable as false.
+const scrapeCraigslist = async (url) => {
+  //TODO: Might want to use something like puppeteer-cluster to cut down on browser instances. Might not matter though.
+  const browser = await puppeteer.launch({});
+  const page = await browser.newPage();
+  const res = await page.goto(url);
+  let productInfo = undefined;
+  if (res.status() == "200") {
+    productInfo = await page.evaluate(() => {
+      const galleryElement = document.querySelector(".gallery");
+      return {
+        title: document.querySelector("#titletextonly").textContent,
+        price: document.querySelector(".price").textContent,
+        imageUrl:
+          //If there isn't a given image, we use the URL of the placeholder image Craigslist uses.
+          galleryElement != null
+            ? document.querySelector(".gallery").children[3].firstElementChild
+                .firstElementChild.firstElementChild.src
```

Comment:
The problem with this approach is that, if the developers at "CraigsList" change the order of any one thing in this series of DOM selections, then you'll get something you didn't intend. Is there no other way of finding the node you're looking for?

Reply:
I don't like to think of it in terms of "acceptable" or not. I just care that:

- It works
- It won't break when someone at craigslist decides to change their UI (because they probably will)

It seems to me more likely that they will change the order of the elements than the id naming conventions they use, so it might be more resilient. But I'll leave it up to you to decide 🙂